lib/template-loader.php

```diff
@@ -416,16 +415,20 @@ function gutenberg_strip_php_suffix( $template_file ) {
  * @return array Filtered editor settings.
  */
 function gutenberg_template_loader_filter_block_editor_settings( $settings ) {
-	global $post, $_wp_current_template_id;
+	global $post;
 
 	if ( ! $post || ! post_type_exists( 'wp_template' ) || ! post_type_exists( 'wp_template_part' ) ) {
 		return $settings;
 	}
 
-	// Create template part auto-drafts for the edited post.
-	$post = isset( $_wp_current_template_id )
-		? get_post( $_wp_current_template_id ) // It's a template.
-		: get_post(); // It's a post.
+	// If this is the Site Editor, auto-drafts for template parts have already been generated
```

Comment:
is this true for all template parts?

Reply:
I think only for template parts corresponding to the loaded template?

I was looking at something like https://github.com/WordPress/gutenberg/pull/23254 to ensure all template parts supplied by a theme have auto-drafts created.

Reply:
All template parts used in resolved templates.

Reply:
do we need to care about non-resolved templates here?

Reply:
No, they are now fetched via REST.